modules/UI/shared_video/SharedVideo.js

```diff
@@ -267,6 +271,13 @@ export default class SharedVideoManager {
 
             VideoLayout.addLargeVideoContainer(
                 SHARED_VIDEO_CONTAINER_TYPE, self.sharedVideo);
+
+            APP.store.dispatch(participantJoined({
+                id: self.url,
+                name: player.getVideoData().title,
+                isSharedVideo: true
+            }));
```

Comment:
Here we fake the youtube video as a user. This is so pinning behavior and future filmstrip behavior can treat the shared video the same as other participants. This seems like the hackiest part of these changes.